src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -821,6 +830,28 @@ public BigDecimal(char[] in, MathContext mc) {
         this(in, 0, in.length, mc);
     }
 
+    private record CharArraySequence(char[] array, int offset, int length) implements CharSequence {
+        CharArraySequence {
+            if ((offset | length) < 0 || offset + length > array.length)
+                throw new NumberFormatException();
+        }
+
+        @Override
+        public int length() {
+            return length;
+        }
+
+        @Override
+        public char charAt(int offset) {
+            return array[offset];
```

Comment:
Good refactor and correctness fix to change this to `offset, length`, but this method looks incorrect now. The parameter should be the `offset`-relative index, no?